### PR TITLE
IDENTITY-4019 Admin service methods needed in UserAccountConnector and UserAccountAssociationService

### PR DESCRIPTION
**Pull request by cdwijayarathna**

### CODE REVIEW

File: components/user-mgt/org.wso2.carbon.identity.user.account.association/src/main/java/org/wso2/carbon/identity/user/account/association/UserAccountAssociationService.java [hunk 1]
```diff
@@ -38,14 +48,191 @@
...
+    public void createUserAccountAssociation(String userName, char[] password) throws
+            UserAccountAssociationClientException {
+
+        String loggedInUser = UserCoreUtil.addTenantDomainToEntry(CarbonContext.getThreadLocalCarbonContext()
+                .getUsername(), CarbonContext.getThreadLocalCarbonContext().getTenantDomain());
+
+        org.wso2.carbon.user.api.UserRealm userRealm = null;
+        RealmService realmService = null;
+        String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(userName);
+        int tenantId = -1;
```
Comment (johannnallathamby) [inline components/user-mgt/org.wso2.carbon.identity.user.account.association/src/main/java/org/wso2/carbon/identity/user/account/association/UserAccountAssociationService.java]: Use MultitenantConstants.INVALID_TENANT_ID

File: components/user-mgt/org.wso2.carbon.identity.user.account.association/src/main/java/org/wso2/carbon/identity/user/account/association/UserAccountAssociationService.java [hunk 2]
```diff
@@ -58,12 +243,153 @@ public void createUserAccountAssociation(String userName, char [] password) thro
...
+
+                }
+                throw new UserAccountAssociationClientException(UserAccountAssociationConstants
+                        .ErrorMessages.INVALID_ASSOCIATION.toString());
+            }
+            UserAccountConnectorImpl.getInstance().deleteUserAccountAssociation(userName);
+        } catch (UserAccountAssociationServerException e) {
+            log.error(UserAccountAssociationConstants.ErrorMessages.CONN_DELETE_ERROR.getDescription(), e);
+            throw new UserAccountAssociationClientException(UserAccountAssociationConstants.ErrorMessages
+                    .CONN_DELETE_ERROR.toString());
```
Comment (kasungayan) [inline components/user-mgt/org.wso2.carbon.identity.user.account.association/src/main/java/org/wso2/carbon/identity/user/account/association/UserAccountAssociationService.java]: Is there any specific reason for doing both logging the exception and throwing ?

File: components/user-mgt/org.wso2.carbon.identity.user.account.association/src/main/java/org/wso2/carbon/identity/user/account/association/UserAccountAssociationService.java [hunk 3]
```diff
@@ -58,12 +243,153 @@ public void createUserAccountAssociation(String userName, char [] password) thro
...
+     * @throws org.wso2.carbon.identity.user.account.association.exception.UserAccountAssociationClientException
+     */
+    public void deleteAssociationsOfUser(String userName) throws UserAccountAssociationClientException {
+
+        try {
+            UserAccountConnectorImpl.getInstance().deleteUserAccountAssociation(userName);
+        } catch (UserAccountAssociationServerException e) {
+            log.error(UserAccountAssociationConstants.ErrorMessages.CONN_DELETE_ERROR.getDescription(), e);
+            throw new UserAccountAssociationClientException(UserAccountAssociationConstants.ErrorMessages
+                    .CONN_DELETE_ERROR.toString());
```
Comment (kasungayan) [inline components/user-mgt/org.wso2.carbon.identity.user.account.association/src/main/java/org/wso2/carbon/identity/user/account/association/UserAccountAssociationService.java]: Is there any specific reason for doing both logging the exception and throwing ?

File: components/user-mgt/org.wso2.carbon.identity.user.account.association/src/main/java/org/wso2/carbon/identity/user/account/association/UserAccountAssociationService.java (hunk 2 above)
Comment (cdwijayarathna) [inline components/user-mgt/org.wso2.carbon.identity.user.account.association/src/main/java/org/wso2/carbon/identity/user/account/association/UserAccountAssociationService.java]: This is a admin service, here I have logged server exception without sending it to the client side and throw a client exception with user friendly error code and message.

File: components/user-mgt/org.wso2.carbon.identity.user.account.association/src/main/java/org/wso2/carbon/identity/user/account/association/UserAccountAssociationService.java (hunk 3 above)
Comment (cdwijayarathna) [inline components/user-mgt/org.wso2.carbon.identity.user.account.association/src/main/java/org/wso2/carbon/identity/user/account/association/UserAccountAssociationService.java]: This is a admin service, here I have logged server exception without sending it to the client side and throw a client exception with user friendly error code and message.

**Pull request merged**